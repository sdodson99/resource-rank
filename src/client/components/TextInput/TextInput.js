import React from 'react';
import PropTypes from 'prop-types';
import styles from './TextInput.module.css';
import { Error } from '@material-ui/icons';

const TextInput = React.forwardRef(function WrappedTextInput(
  { name, label, errorMessage, ...others },
  ref
) {
  const calculateInputClassName = () => {
    let className = 'mt-2 form-control flex-grow';

    if (errorMessage) {
      className += ` ${styles.ErrorTextInput}`;
    }

    return className;
  };

  return (
    <div className={styles.TextInput} data-testid="TextInput">
      <div className="flex flex-col">
        {label && <label htmlFor={name}>{label}</label>}

        <input
          ref={ref}
          name={name}
          className={calculateInputClassName()}
          type="text"
          {...others}
        />

        {errorMessage && (
          <div className="mt-2 error-text text-sm flex items-center">
            <div>
              <Error />
            </div>
            <div className="ml-2">{errorMessage}</div>
          </div>
        )}
      </div>
    </div>
  );
});

TextInput.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  errorMessage: PropTypes.string,
};

TextInput.defaultProps = {};

export default TextInput;
