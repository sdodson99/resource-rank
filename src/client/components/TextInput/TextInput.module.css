.TextInput {
}

input[type='text'].ErrorTextInput {
  @apply border-red-400 outline-none;
}

input[type='text'].ErrorTextInput:focus {
  @apply ring ring-red-50;
}
