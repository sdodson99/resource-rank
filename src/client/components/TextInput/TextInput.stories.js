import React from 'react';
import TextInput from './TextInput';

export default {
  component: TextInput,
  title: 'Components/TextInput',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <TextInput {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Error = Template.bind({});
Error.args = {
  errorMessage: 'Please enter information.',
};
