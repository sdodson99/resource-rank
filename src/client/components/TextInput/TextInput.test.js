import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TextInput from './TextInput';
import renderer from 'react-test-renderer';

describe('<TextInput />', () => {
  it('should mount', () => {
    render(<TextInput name={'input'} />);

    const textInput = screen.getByTestId('TextInput');

    expect(textInput).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<TextInput name={'input'} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correct when label provided', () => {
    const tree = renderer
      .create(<TextInput name={'input'} label={'Data Input'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when error message provided', () => {
    const tree = renderer
      .create(<TextInput name={'input'} errorMessage={'Error message.'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
