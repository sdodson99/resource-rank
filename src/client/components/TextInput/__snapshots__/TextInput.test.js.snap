// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render correct when label provided 1`] = `
<div
  className="TextInput"
  data-testid="TextInput"
>
  <div
    className="flex flex-col"
  >
    <label
      htmlFor="input"
    >
      Data Input
    </label>
    <input
      className="mt-2 form-control flex-grow"
      id="input"
      name="input"
      type="text"
    />
  </div>
</div>
`;

exports[`<TextInput /> should render correctly 1`] = `
<div
  className="TextInput"
  data-testid="TextInput"
>
  <div
    className="flex flex-col"
  >
    <input
      className="mt-2 form-control flex-grow"
      id="input"
      name="input"
      type="text"
    />
  </div>
</div>
`;

exports[`<TextInput /> should render correctly when error message provided 1`] = `
<div
  className="TextInput"
  data-testid="TextInput"
>
  <div
    className="flex flex-col"
  >
    <input
      className="mt-2 form-control flex-grow ErrorTextInput"
      id="input"
      name="input"
      type="text"
    />
    <div
      className="mt-2 error-text text-sm flex items-center"
    >
      <div>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
      </div>
      <div
        className="ml-2"
      >
        Error message.
      </div>
    </div>
  </div>
</div>
`;
